web/src/components/modal/VersionHistoryViewerModal.tsx

```diff
@@ -0,0 +1,116 @@
+  const { dialogClasses, onCancel, onAnimEnd, modalRef, onClick } = useModal({ open, onClose })
+
+  const latestVersion = dataList.at(-1)?.version
+
+  const navigateToVersion = (item: BaseOut) => {
+    navigate(contentPagePath(item.exam, contentType, item.id, item.version))
+    onClose()
+  }
+
+  const update = async (data: BaseOut) => {
+    if (isAssignment(data)) {
+      await updateAssignment(data.id, data)
+    } else if (isInstruction(data)) {
+      const attachmentsToUpdate: AttachmentData[] = data.attachments.map((attachment) => ({
+        name: attachment.name,
+        attachment: attachment,
+        language: attachment.language.toLowerCase() as AttachmentLanguage
+      }))
+      await createNewVersionInstruction(data.id, data, attachmentsToUpdate, [])
+    } else if (isCertificate(data)) {
+      await updateCertificate(data.id, data, null, null)
+    }
+    onClose(true)
+  }
+
+  return (
+    <dialog
+      ref={modalRef}
+      className={dialogClasses}
+      onClose={() => onClose()}
+      onCancel={onCancel}
+      onClick={onClick}
+      onAnimationEnd={onAnimEnd}
+      aria-modal="true"
+      data-testid="version-history-modal">
+      <div className={styles['modal__container']}>
+        <ModalHeader modalTitle={t('version-control.muokkaushistoria')} onClick={onClose} />
+
+        <div className="h-auto min-h-[20rem] max-h-[30rem] overflow-scroll px-6 py-2">
+          <table className="w-full text-left">
+            <thead className="border-b border-gray-border">
+              <tr>
+                <th className="font-semibold pr-6 py-1">{t('version-control.versio')}</th>
+                <th className="font-semibold pr-6">{t('version-control.muokattu')}</th>
+                <th className="font-semibold pr-6">{t('version-control.muokkaaja')}</th>
+                <th className="font-semibold pr-6">{t('version-control.toiminnot')}</th>
+              </tr>
+            </thead>
+            <tbody>
+              {dataList.toReversed().map((item) => (
+                <tr className="even:bg-gray-bg" key={item.version} data-testid={`version-history-item-${item.version}`}>
+                  <td className="pr-6 py-3 pl-3">{item.version}</td>
+                  <td className="pr-6">{toLocaleDate(item.updatedAt)}</td>
+                  {/*todo: hae päivittäjän nimi */}
+                  <td className="pr-6">{item.updaterOid.slice(0, 10)}</td>
+                  <td className="pr-6 w-auto whitespace-nowrap">
+                    {latestVersion === item.version ? (
+                      t('version-control.viimeisin-versio')
+                    ) : (
+                      <div className="row gap-2 flex-wrap">
+                        <Button
+                          variant="buttonGhost"
+                          customClass="p-0"
+                          onClick={() => navigateToVersion(item)}
+                          data-testid={`version-history-show-item-${item.version}`}>
```

Comment:
Ehdotan, että nää `version-history-item-${item.version}` sisällä olevat testid:t ei sisältäis enää indeksiä vaan tyyliin pelkkä `show`. Se löytyis näin: `page.getByTestId('version-history-item-1').getByTestId('show')

Huono puoli: tarttee 2 getByTestId:tä
Hyvä puoli: paljon lyhyemmät testid:t itemin sisällä eikä tartte tunkee indeksiä moneen paikkaan.

Paavon POM-ohjeissa muistaakseni suositeltiin tätä tyyliä.

Reply:
Joo toi tosi hyvä!